src/ints/mouse.cpp

```diff
@@ -18,1549 +18,932 @@
+    uint8_t start_delay_dos_mov = 5; // for DOS move/wheel events
+
+private:
+    MouseQueue(const MouseQueue &)            = delete;
+    MouseQueue &operator=(const MouseQueue &) = delete;
+
+    void AggregateEventsDOS(MouseEvent &event);
+    MouseEvent &PopEventBtn();
+    void UpdateDelayCounters();
+
+    std::array<MouseEvent, 8> events = {}; // a modulo queue of events
+    uint8_t idx_first                = 0;  // index of the first event
+    uint8_t num_events               = 0;  // number of events in the queue
+
+    MouseButtons12S last_buttons_12S = 0; // last buttons reported
+
+    bool queue_overflow    = false;
+    bool timer_in_progress = false;
+
+    // Time in milliseconds which has to elapse before event can take place
+    uint8_t delay_ps2     = 0;
+    uint8_t delay_dos_btn = 0; // for DOS button events
+    uint8_t delay_dos_mov = 0; // for DOS move/wheel events
+
+    // Events for which a flag is enough to store them
+    bool event_ps2       = false;
+    bool event_dos_moved = false;
+    bool event_dos_wheel = false;
+
+    bool prefer_ps2 = false; // true = next time prefer PS/2 event
+
+    uint32_t ticks_start = 0; // PIC_Ticks value when timer starts
+
+    // Helper masks for aggregating DOS button events
+    static constexpr uint8_t aggr_mask_pressed =
+            static_cast<uint8_t>(MouseEventId::PressedLeft)  |
+            static_cast<uint8_t>(MouseEventId::PressedRight) |
+            static_cast<uint8_t>(MouseEventId::PressedMiddle);
+    static constexpr uint8_t aggr_mask_released =
+            static_cast<uint8_t>(MouseEventId::ReleasedLeft)  |
+            static_cast<uint8_t>(MouseEventId::ReleasedRight) |
+            static_cast<uint8_t>(MouseEventId::ReleasedMiddle);
+
+    // Helpers to check if there are events in the queue
+    inline bool HasEventDOSmov() const
+    {
+        return event_dos_moved || event_dos_wheel;
+    }
+    inline bool HasEventDOSbtn() const
+    {
+        return (num_events != 0);
+    }
+    inline bool HasEventDOSany() const
+    {
+        return HasEventDOSmov() || HasEventDOSbtn();
+    }
+    inline bool HasEventPS2() const
+    {
+        return event_ps2;
+    }
+    inline bool HasEventAny() const
+    {
+        return HasEventDOSany() || HasEventPS2();
+    }
+
+    // Helpers to check if there are events ready to be handled
+    inline bool HasReadyEventPS2() const
+    {
+        return HasEventPS2() && !delay_ps2;
+    }
+    inline bool HasReadyEventDOSmov() const
+    {
+        return HasEventDOSmov() && !delay_dos_mov &&
+               !mouse_shared.dos_cb_running; // callback busy = no new
+                                             // event
+    }
+    inline bool HasReadyEventDOSbtn() const
+    {
+        return HasEventDOSbtn() && !delay_dos_btn &&
+               !mouse_shared.dos_cb_running; // callback busy = no new
+                                             // event
+    }
+    inline bool HasReadyEventAny() const
```

Comment:
I mentioned this before, and I still think the `inline` is mostly a feel-good thing for the person who wrote the code 😄 If the compiler cannot figure out it's worthwhile to inline these simple expressions, it is screwed in a major way anyway... (so I'd get rid of them all, the compiler surely can figure something dead simple like this out by itself these days)